containers/java-postgres/test-project/src/test/java/com/mycompany/app/AppTest.java

```diff
@@ -0,0 +1,125 @@
+/*--------------------------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+ *-------------------------------------------------------------------------------------------------------------*/
+
+package com.mycompany.app;
+
+import org.junit.Test;
+
+import junit.framework.AssertionFailedError;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+public class AppTest
+{
+
+
+    public static boolean pingAddress(String ipAddress) throws UnknownHostException, IOException
+    {
+        InetAddress postgresAddress = InetAddress.getByName(ipAddress);
+        System.out.println("Sending Ping Request to " + ipAddress);
+        if (postgresAddress.isReachable(5000)){
+            System.out.println("Successfully Reached: " + ipAddress);
+            return true;
+        }
+        System.out.println("Could not reach or connect to: " + ipAddress);
+        return false;
+    }
+
+    public static boolean dbLogin(String ipAddress, String username, String password){
+        Connection c = null;
+        try {
+            Class.forName("org.postgresql.Driver");
+            c = DriverManager
+               .getConnection("jdbc:postgresql://postgresdb/postgres",
+               "postgres", "postgres");
+            return true;
+         } catch (Exception e) {
+            e.printStackTrace();
+            System.err.println(e.getClass().getName()+": "+e.getMessage());
+            return false;
+         }
+    }
+
+    public static boolean listDatabases(String ipAddress, String username, String password){
+        Connection c = null;
+        Statement stmt = null;
+
+        try {
+            Class.forName("org.postgresql.Driver");
+            c = DriverManager
+               .getConnection("jdbc:postgresql://postgresdb/postgres",
+               "postgres", "postgres");
+            c.setAutoCommit(false);
+            stmt = c.createStatement();
+            ResultSet rs = stmt.executeQuery( "select * from pg_database;" );
+            System.out.println("List of databases in this cluster.");
+            while (rs.next()){
```

Comment:
Suggestion: Instead of iterating the result, perhaps with just one call to 'poke' the db would be enough.

Reply:
Well, we're poking the DB with the login anyway and testing the container. This way we can iterate from an actual result set right?

Reply:
Yes, but my comment is more towards overhead of iterating and printing the content.